db.go

```diff
@@ -111,13 +95,38 @@ func (db *DB) doCompaction() {
 
 func (db *DB) checkAndTriggerCompaction() bool {
 	readyToExit := true
-	for idx, level := range db.levels {
-		if len(level.sstables) > maxLevelSSTables[idx] {
+
+	if db.needL0Compaction() {
+		db.compactionChan <- 0
+		readyToExit = false
+	}
+
+	if !db.needLevelNCompactions() {
+		readyToExit = false
+	}
+
+	return readyToExit
+}
+
+func (db *DB) needL0Compaction() bool {
+	return len(db.levels[0].sstables) > l0Capacity
```

Comment:
### Optimizations

- **Redundant Function Calls**: In `needL0Compaction`, the length of `db.levels[0].sstables` is computed twice. Cache this value in a variable to avoid redundant computations:

```go
numSSTables := len(db.levels[0].sstables)
return numSSTables > l0Capacity
```